CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 Hennadii Stepanov
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+cmake_minimum_required(VERSION 3.1)
+if(CMAKE_VERSION VERSION_GREATER 3.14)
+  # MSVC warning flags are not in CMAKE_<LANG>_FLAGS by default.
+  # See: https://cmake.org/cmake/help/latest/policy/CMP0092.html
+  cmake_policy(SET CMP0092 NEW)
+endif()
+project(secp256k1 VERSION 0.1.0 LANGUAGES C)
+
+set(CMAKE_C_STANDARD 90)
+set(CMAKE_C_EXTENSIONS OFF)
+
+option(ENABLE_DEV_MODE "enable all binaries and modules by default but individual options can still be overridden explicitly" OFF)
```

Comment:
Hm, I think having this as an option doesn't really work with CMake's "cache-based" config system. When `ENABLE_MODULE_RECOVERY` is `OFF` in the cache, setting `ENABLE_DEV_MODE` won't enable it. 

Also there's no "order" on the config options. In ./configure, we process options from left-to-right, so later options can override the dev mode defaults. But this doesn't make sense in CMake. 

Maybe there's another way to express this . https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html looks promising. Though this could be done in a separate PR. dev-mode is not a crucial feature and anyway we'd still have autotools.

Reply:
The `ENABLE_DEV_MODE` option has been [removed](https://github.com/bitcoin-core/secp256k1/pull/1113#issuecomment-1179331707).